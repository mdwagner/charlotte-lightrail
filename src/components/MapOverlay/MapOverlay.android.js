import React from 'react';
import PropTypes from 'prop-types';
import { Alert, Image } from 'react-native';
import defaults from 'react-native-default-preference';
import SegmentedControlTab from 'react-native-segmented-control-tab';
import LocationButton from '../LocationButton';
import StationSlider from '../StationSlider';
import {
  AboutTouchableOpacity, ContainerView, HeaderContainerView,
  HeaderView, HelpView, OfflineButtonTouchableOpacity, OfflineText
} from './MapOverlayCss';

export default class MapOverlay extends React.Component {

  static propTypes = {
    connected: PropTypes.bool.isRequired,
    error: PropTypes.string,
    fetchNearest: PropTypes.func.isRequired,
    loading: PropTypes.bool.isRequired,
    locationDenied: PropTypes.bool.isRequired,
    mode: PropTypes.string.isRequired, // eslint-disable-line
    nearestStationIndex: PropTypes.number, // eslint-disable-line
    seeAllStations: PropTypes.func.isRequired,
    showCallout: PropTypes.func.isRequired, // eslint-disable-line
    stationDistances: PropTypes.array, // eslint-disable-line
    navigation: PropTypes.shape({
      navigate: PropTypes.func.isRequired
    })
  }

  state = {
    displaySchedule: false,
    selectedIndex: 1
  }

  handleZeroState = () => {
    const { fetchNearest, locationDenied } = this.props;
    if (locationDenied) {
      Alert.alert('Enable location services in Privacy Settings to see your location and nearest station');
    } else {
      fetchNearest();
    }
  }

  handleIndexChange = (index) => {
    const { fetchNearest } = this.props;
    let value;
    if (index === 0) {
      value = 'driving';
    } else {
      value = 'walking';
    }
    defaults.set('SavedDirectionsChoice', JSON.stringify(value)).catch(err => console.log(err));
    fetchNearest(value);
    this.setState({
      ...this.state,
      selectedIndex: index
    });
  }

  render() {
    const { displaySchedule } = this.state;
    const { connected, error, loading, locationDenied, seeAllStations, mode, fetchNearest } = this.props;
    const { navigate } = this.props.navigation;

    const failText = locationDenied ? 'Location Disabled - Enable' : 'Offline Mode - Try Again';

    return (
      <ContainerView pointerEvents="box-none">
        {!displaySchedule &&
          <HeaderView pointerEvents="box-none">
            <HeaderContainerView>
              <AboutTouchableOpacity onPress={() => navigate('Faq')}>
                <HelpView>
                  <Image
                    // eslint-disable-next-line
                    source={require('../../assets/icons/help/ic_help_white.png')}
                  />
                </HelpView>
              </AboutTouchableOpacity>
              {connected && !locationDenied
                ?
                  <SegmentedControlTab
                    values={['Driving', 'Walking']}
                    selectedIndex={mode === 'driving' ? 0 : 1}
                    onValueChange={(value) => {
                      defaults.set('SavedDirectionsChoice', value.toLowerCase()).catch(err => console.log(err));
                      fetchNearest(value.toLowerCase());
                    }}
                    tintColor="white"
                  />
                :
                  <OfflineButtonTouchableOpacity onPress={() => this.handleZeroState()}>
                    <OfflineText allowFontScaling={false}>{failText}</OfflineText>
                  </OfflineButtonTouchableOpacity>
              }
              <LocationButton seeAllStations={seeAllStations} error={error} loading={loading} />
            </HeaderContainerView>
          </HeaderView>
        }
        <StationSlider {...this.props} />
      </ContainerView>
    );
  }
}
