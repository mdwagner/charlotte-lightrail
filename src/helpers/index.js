export { withHelpers, withStaticData } from './hoc';