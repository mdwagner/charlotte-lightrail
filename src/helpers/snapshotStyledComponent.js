import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import { COLORS } from 'assets/styles/constants';
import renderer from 'react-test-renderer';

export const snapshotStyledComponent = (component) => {
  const snapshot = renderer.create(
    <ThemeProvider theme={COLORS}>
      {component}
    </ThemeProvider>
  ).toJSON()
  expect(snapshot).toMatchSnapshot();
};