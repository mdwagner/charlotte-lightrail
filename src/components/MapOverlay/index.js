import { Platform } from 'react-native';
import { default as IOS } from './MapOverlay.ios';
import { default as ANDROID } from './MapOverlay.android';

const exportedComponent = Platform.OS === 'ios' ? IOS : ANDROID;
export default exportedComponent;
