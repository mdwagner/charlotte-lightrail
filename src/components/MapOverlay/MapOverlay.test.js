import React from 'react';
import { snapshotStyledComponent } from '../../helpers/snapshotStyledComponent';
import MapOverlay from './MapOverlay';
jest.mock('SegmentedControlIOS');
import renderer from 'react-test-renderer';

describe('<MapOverlay />', () => {
  let props;

  beforeAll(() => {
    props = {
      connected: true,
      loading: false,
      mode: 'walking',
      fetchNearest: jest.fn(),
      showCallout: jest.fn(),
      locationDenied: false,
      seeAllStations: jest.fn(),
      navigation: { navigate: jest.fn() }
    }
  });

  it('renders correctly', () => {
    snapshotStyledComponent(
      <MapOverlay {...props} />
    );
  });
});