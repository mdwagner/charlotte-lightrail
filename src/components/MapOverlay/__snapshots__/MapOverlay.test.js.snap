// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapOverlay /> renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      undefined,
    ]
  }
>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "backgroundColor": "rgba(1, 42, 60, 0.97)",
          "height": 40,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexBasis": 0,
            "flexGrow": 0.8,
            "flexShrink": 1,
            "opacity": 1,
            "paddingLeft": 14,
            "paddingTop": 1,
          }
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              undefined,
            ]
          }
        >
          <Image
            source={1}
          />
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "flexBasis": 0,
            "flexGrow": 0.8,
            "flexShrink": 1,
            "opacity": 1,
            "paddingRight": 14,
            "paddingTop": 1,
          }
        }
        testID={undefined}
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "opacity": 1,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "width": 375,
          },
          undefined,
        ]
      }
    >
      <ScrollView
        automaticallyAdjustContentInsets={false}
        horizontal={true}
        onMomentumScrollEnd={[Function]}
        pagingEnabled={true}
      >
        <View
          style={
            Array [
              Object {
                "height": 279,
                "width": 375,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderBottomColor": "rgba(0, 20, 30, 0.8)",
                  "borderBottomWidth": 25,
                  "borderLeftColor": "transparent",
                  "borderLeftWidth": 25,
                  "borderRightColor": "transparent",
                  "borderRightWidth": 25,
                  "borderStyle": "solid",
                  "flexDirection": "row",
                  "height": 0,
                  "width": 0,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 20, 30, 0.8)",
                  "height": 254,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="large"
            />
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#CCC",
                    "fontSize": 16,
                    "paddingVertical": 10,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Finding nearest station...
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 279,
                "width": 375,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderBottomColor": "rgba(0, 20, 30, 0.8)",
                  "borderBottomWidth": 25,
                  "borderLeftColor": "transparent",
                  "borderLeftWidth": 25,
                  "borderRightColor": "transparent",
                  "borderRightWidth": 25,
                  "borderStyle": "solid",
                  "flexDirection": "row",
                  "height": 0,
                  "width": 0,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 20, 30, 0.8)",
                  "height": 254,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="large"
            />
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#CCC",
                    "fontSize": 16,
                    "paddingVertical": 10,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Finding nearest station...
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 279,
                "width": 375,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderBottomColor": "rgba(0, 20, 30, 0.8)",
                  "borderBottomWidth": 25,
                  "borderLeftColor": "transparent",
                  "borderLeftWidth": 25,
                  "borderRightColor": "transparent",
                  "borderRightWidth": 25,
                  "borderStyle": "solid",
                  "flexDirection": "row",
                  "height": 0,
                  "width": 0,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 20, 30, 0.8)",
                  "height": 254,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="large"
            />
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#CCC",
                    "fontSize": 16,
                    "paddingVertical": 10,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Finding nearest station...
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 279,
                "width": 375,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderBottomColor": "rgba(0, 20, 30, 0.8)",
                  "borderBottomWidth": 25,
                  "borderLeftColor": "transparent",
                  "borderLeftWidth": 25,
                  "borderRightColor": "transparent",
                  "borderRightWidth": 25,
                  "borderStyle": "solid",
                  "flexDirection": "row",
                  "height": 0,
                  "width": 0,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 20, 30, 0.8)",
                  "height": 254,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="large"
            />
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#CCC",
                    "fontSize": 16,
                    "paddingVertical": 10,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Finding nearest station...
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 279,
                "width": 375,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderBottomColor": "rgba(0, 20, 30, 0.8)",
                  "borderBottomWidth": 25,
                  "borderLeftColor": "transparent",
                  "borderLeftWidth": 25,
                  "borderRightColor": "transparent",
                  "borderRightWidth": 25,
                  "borderStyle": "solid",
                  "flexDirection": "row",
                  "height": 0,
                  "width": 0,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 20, 30, 0.8)",
                  "height": 254,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="large"
            />
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#CCC",
                    "fontSize": 16,
                    "paddingVertical": 10,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Finding nearest station...
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 279,
                "width": 375,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderBottomColor": "rgba(0, 20, 30, 0.8)",
                  "borderBottomWidth": 25,
                  "borderLeftColor": "transparent",
                  "borderLeftWidth": 25,
                  "borderRightColor": "transparent",
                  "borderRightWidth": 25,
                  "borderStyle": "solid",
                  "flexDirection": "row",
                  "height": 0,
                  "width": 0,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 20, 30, 0.8)",
                  "height": 254,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="large"
            />
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#CCC",
                    "fontSize": 16,
                    "paddingVertical": 10,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Finding nearest station...
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 279,
                "width": 375,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderBottomColor": "rgba(0, 20, 30, 0.8)",
                  "borderBottomWidth": 25,
                  "borderLeftColor": "transparent",
                  "borderLeftWidth": 25,
                  "borderRightColor": "transparent",
                  "borderRightWidth": 25,
                  "borderStyle": "solid",
                  "flexDirection": "row",
                  "height": 0,
                  "width": 0,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 20, 30, 0.8)",
                  "height": 254,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="large"
            />
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#CCC",
                    "fontSize": 16,
                    "paddingVertical": 10,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Finding nearest station...
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 279,
                "width": 375,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderBottomColor": "rgba(0, 20, 30, 0.8)",
                  "borderBottomWidth": 25,
                  "borderLeftColor": "transparent",
                  "borderLeftWidth": 25,
                  "borderRightColor": "transparent",
                  "borderRightWidth": 25,
                  "borderStyle": "solid",
                  "flexDirection": "row",
                  "height": 0,
                  "width": 0,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 20, 30, 0.8)",
                  "height": 254,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="large"
            />
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#CCC",
                    "fontSize": 16,
                    "paddingVertical": 10,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Finding nearest station...
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 279,
                "width": 375,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderBottomColor": "rgba(0, 20, 30, 0.8)",
                  "borderBottomWidth": 25,
                  "borderLeftColor": "transparent",
                  "borderLeftWidth": 25,
                  "borderRightColor": "transparent",
                  "borderRightWidth": 25,
                  "borderStyle": "solid",
                  "flexDirection": "row",
                  "height": 0,
                  "width": 0,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 20, 30, 0.8)",
                  "height": 254,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="large"
            />
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#CCC",
                    "fontSize": 16,
                    "paddingVertical": 10,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Finding nearest station...
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 279,
                "width": 375,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderBottomColor": "rgba(0, 20, 30, 0.8)",
                  "borderBottomWidth": 25,
                  "borderLeftColor": "transparent",
                  "borderLeftWidth": 25,
                  "borderRightColor": "transparent",
                  "borderRightWidth": 25,
                  "borderStyle": "solid",
                  "flexDirection": "row",
                  "height": 0,
                  "width": 0,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 20, 30, 0.8)",
                  "height": 254,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="large"
            />
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#CCC",
                    "fontSize": 16,
                    "paddingVertical": 10,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Finding nearest station...
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 279,
                "width": 375,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderBottomColor": "rgba(0, 20, 30, 0.8)",
                  "borderBottomWidth": 25,
                  "borderLeftColor": "transparent",
                  "borderLeftWidth": 25,
                  "borderRightColor": "transparent",
                  "borderRightWidth": 25,
                  "borderStyle": "solid",
                  "flexDirection": "row",
                  "height": 0,
                  "width": 0,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 20, 30, 0.8)",
                  "height": 254,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="large"
            />
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#CCC",
                    "fontSize": 16,
                    "paddingVertical": 10,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Finding nearest station...
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 279,
                "width": 375,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderBottomColor": "rgba(0, 20, 30, 0.8)",
                  "borderBottomWidth": 25,
                  "borderLeftColor": "transparent",
                  "borderLeftWidth": 25,
                  "borderRightColor": "transparent",
                  "borderRightWidth": 25,
                  "borderStyle": "solid",
                  "flexDirection": "row",
                  "height": 0,
                  "width": 0,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 20, 30, 0.8)",
                  "height": 254,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="large"
            />
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#CCC",
                    "fontSize": 16,
                    "paddingVertical": 10,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Finding nearest station...
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 279,
                "width": 375,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderBottomColor": "rgba(0, 20, 30, 0.8)",
                  "borderBottomWidth": 25,
                  "borderLeftColor": "transparent",
                  "borderLeftWidth": 25,
                  "borderRightColor": "transparent",
                  "borderRightWidth": 25,
                  "borderStyle": "solid",
                  "flexDirection": "row",
                  "height": 0,
                  "width": 0,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 20, 30, 0.8)",
                  "height": 254,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="large"
            />
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#CCC",
                    "fontSize": 16,
                    "paddingVertical": 10,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Finding nearest station...
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 279,
                "width": 375,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderBottomColor": "rgba(0, 20, 30, 0.8)",
                  "borderBottomWidth": 25,
                  "borderLeftColor": "transparent",
                  "borderLeftWidth": 25,
                  "borderRightColor": "transparent",
                  "borderRightWidth": 25,
                  "borderStyle": "solid",
                  "flexDirection": "row",
                  "height": 0,
                  "width": 0,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 20, 30, 0.8)",
                  "height": 254,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="large"
            />
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#CCC",
                    "fontSize": 16,
                    "paddingVertical": 10,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Finding nearest station...
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 279,
                "width": 375,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderBottomColor": "rgba(0, 20, 30, 0.8)",
                  "borderBottomWidth": 25,
                  "borderLeftColor": "transparent",
                  "borderLeftWidth": 25,
                  "borderRightColor": "transparent",
                  "borderRightWidth": 25,
                  "borderStyle": "solid",
                  "flexDirection": "row",
                  "height": 0,
                  "width": 0,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 20, 30, 0.8)",
                  "height": 254,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="large"
            />
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#CCC",
                    "fontSize": 16,
                    "paddingVertical": 10,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Finding nearest station...
            </Text>
          </View>
        </View>
      </ScrollView>
    </View>
    <View
      style={
        Array [
          Object {
            "bottom": 258,
            "flexDirection": "row",
            "position": "absolute",
            "width": 375,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 20,
              "width": 20,
            },
            undefined,
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAIIAAAAoCAMAAAAFWtJHAAAAwFBMVEUAAAAAAAAAAABtbW0AAAAAAAAAAAAAAAAAAAAAAAClpaUAAADp6ekAAAD5+fna2toAAAAMDAzv7+/Nzc0AAAA2Njb8/Pz9/f3T09MAAAAAAAD7+/sAAAArKyuxsbH39/fs7OwbGxuIiIjz8/N8fHyenp7u7u74+PgAAAC8vLxWVlbx8fF1dXXl5eVcXFyUlJTQ0NDFxcVCQkLAwMC4uLj19fXo6OjW1tarq6ve3t77+/vi4uL6+vrKysrNzc3///8w7gSSAAAAP3RSTlMAOQNdPSYBPywKexLLGPCxNEHXnzFL+v2nGwf1IEiE6dBFad9jd9PuLo1V2mDDV3Cjl06SiuXIq4C3973ym6BQMVUPAAAEXElEQVR4Ae2WCVP6OBiH05L0l1IqrVbkKHJ54I0oHn+PfP9vtUle0z/YdhbH2XVnd58ZnRJIeHiPJOx//mH4vQSAN+8FjAhFxgHIaPvJeZ99hxwEElon5iAQbj85Y98g8ODwjEOMAvGFyeE3FEKgodTBqj0BJGN9DhyNd5Ta3ean9QEopfaA+LsKhnEKRExqg4FSP6Og7oEkAjBWnxSCgBX4xF+kcLoPcOBQrSv0e5kH7s1j37jECQieCTPiFGxL5VHw2zQWCeeJiPt6kjRQw0XSkIdVChf67xGa4alSnZlT6HEQ8CK9ANbhvXUF9xlDkBfTuHDWScgC9+z5FQpPI12TlwC6+sV7ixR8CUMKiwjm2GQeOQWHMGuHGdbnObJAwCEqFJpNU5H6uaPUaEIKiQfg+PHk1+u4OwW9PlWW2ctbA4BHCtp+cNK+H8Jos4gDmC5ar4Nx9waaG/2B13NgDqS7+vm2RgEtEws82P+kwIHhs/pgkQKcFIhfd7CogtGNjYMHTLpurD0ERbYFw4JaD3GlQuNAL/JEsSAF4HqlCnaHACk4WhOn4OgCkMD5hSpYNYDJTD8Y46n+jsE1kPhVCuR6QBXhFK7MUOu9O6b1SWF3b+/9ZVWMGOlu93E8UDaAhgc7bfH+0DHqKXCkHzoNDFfU+zxiVQrUC9QXTuHYtKpN59OA3IxCG4b7jh6ZFuVockaNTW09mkJzOaPU49a6mE9cAchZpQJNpUWcwgV9r6FJswsFKrITp2B5pMBMdnS0z2HZNy2+BNKxSZxZfglkrFYBJxQnpzA5sN/HheR2aFQoZBLAi149dQoyAYYjW0hHlHguBAdMcR0DuDZ5omevX6+AI8qcU7ikKT3GBHCnXwydgmCC0tRwCnGQ2Wp6Be71yNIWfQSkOl9vAI1SBCNWrwC01RROgX7BuT2HI4r7tFAw086p/NwZEdOEa7R1uAFuNmQPuKAEAjYNQ0CyeoUEWHYBnpQVQgpvc0Ph+gsKlAnKg1+vEHsw5LKciLKCAJobiWBzYFGbCKpHqkZZrxBFHEASyFI59vJPCskcwNVGOWZAOqsrR+pKbaNeAMT1CixMEtlnsqopNxUMzVJT3tY35aXZm6a6Y9QhwMN6BUJWbE1lhbMO1WehkO7poO0sK7em9MJGxp1XSbC1gtugzzSLQmGsX7VntJGSwsPZ2d2z3bIPKzdoOp3Wzqt8G4XyMVUoFIxLx1S7+piaHtCvR3FeRVsq0GFdp9C5TbGpcNqsPqyHKxcfd14h21KhuLKUFU4f3osrC7F6uV3WXFnadL7wyAPeKDXw2RoJCO5GY4DouYvb/gepVXheLoewzPseQG9N/vzilrMIjoStE3++zvle4eSurw7XEe76ynI4aq+v7lEyt1x5awiFlFLQbHKIpabnM3eJLym4Szzzc/du7SU+zOXv9UNpECH7IoH/gecURPlN9vdQpeD47yhIFNX0U0QgvID9nENm+yxk/xb+AGAjNfRZuk9qAAAAAElFTkSuQmCC",
            }
          }
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "height": 20,
                "resizeMode": "contain",
                "width": 50,
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 20,
            "marginHorizontal": 4,
            "opacity": 1,
            "position": "relative",
            "width": 20,
          }
        }
        testID={undefined}
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 20,
                "width": 20,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
